src/TaskScheduler.h

```diff
@@ -562,6 +563,26 @@ inline void Scheduler::init() {
 #endif  // _TASK_SLEEP_ON_IDLE_RUN
 }
 
+/** Check if task aTask has been added to the execution chain.
+ * Returns a boolean true to indicate that the task is in the chain.
+ * @param &aTask - reference to the Task to be checked.
+ */
+inline bool Scheduler::isTaskAdded(Task& aTask) {
+	aTask.iScheduler = this;
+	if (iFirst == NULL)
+		return false; // No task in the list
+
+	const Task *evaluated_task = iFirst;
+	while (evaluated_task->iNext != NULL) {
```

Comment:
I think this loop runs 1 iteration too little, when `aTask` points to the last task in the list (which has `->iNext == NULL`), then this function will never return true. Could probably be fixed by just doing `while (evaluated_task != NULL)`, which also allows removing the `if (iFirst == NULL)` check above.

However, looking at `deleteTask`, I see that this is really a doubly-linked list (i.e. `iNext` and `iPrev` pointers), and that the pointers are cleared when deleting a task, so I think the entire function could just be: `return &aTask == iFirst || aTask.iPrev != NULL;`. This uses the fact that `iPrev` is non-null when the the task is in the list, *except* when it is the first element in the list, so that is checked separately.

Reply:
Good points, that definitely simplifies things and removes a while loop. I pushed the changes already.